src/coreclr/jit/morph.cpp

```diff
@@ -3255,6 +3244,12 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
                         passingSize  = genTypeSize(structBaseType);
                     }
                 }
+#if !defined(TARGET_X86)
+                else
+                {
+                    hasMultiregStructArgs = true;
```

Comment:
Why is this needed here now?

Reply:
It used to be set below:
```c
#if FEATURE_MULTIREG_ARGS
        if (isStructArg)
        {
            if (((arg.AbiInfo.NumRegs + arg.AbiInfo.GetStackSlotsNumber()) > 1) ||
                (isHfaArg && argx->TypeGet() == TYP_STRUCT))
            {
                hasMultiregStructArgs = true;
            }
        }
```
So this is just moving it to here. The benefit it this location is that we don't need to invoke multi-reg morphing if we have already transformed everything into `FIELD_LIST`s.